lib/util.js

```diff
@@ -174,40 +324,62 @@ const updateDBForCRXFile = (endpoint, region, crxFile, componentId, contentHash)
 }
 
 // Update incrementVersion if this code is updated
-const getPreviousVersion = (version) => {
+const getPreviousVersion = (version, diff = 1) => {
   if (version === FirstVersion) {
     // This is the first version
     return FirstVersion
   }
   const versionParts = version.split('.')
-  versionParts[versionParts.length - 1]--
+  versionParts[versionParts.length - 1] -= diff
   return versionParts.join('.')
 }
 
-const uploadExtension = async (name, id, version, crxFile) => {
-  const S3_EXTENSIONS_BUCKET = process.env.S3_EXTENSIONS_BUCKET || 'brave-core-ext'
-  const componentFilename = `extension_${version.replace(/\./g, '_')}.crx`
-  const componentName = `${name.replace(/\s/g, '-')}`
-  const previousComponentFilename = `extension_${getPreviousVersion(version).replace(/\./g, '_')}.crx`
-  // See tag restrictions here: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html#tag-restrictions
-  const componentNameTag = `${componentName.replace(/[^a-zA-Z0-9+\-=._:/@]+/g, '')}`
-  const latestVersionTagKey = 'version'
-  const latestVersionTagValue = `${componentNameTag}/latest`
+const uploadToS3 = async (file, bucket, key, contentType) => {
```

Comment:
`bucket` argument here, but `Bucket: S3_EXTENSIONS_BUCKET` and `s3://${S3_EXTENSIONS_BUCKET}/${key}` hardcoded below